code/controllers/subsystem.dm

```diff
@@ -1,88 +1,91 @@
-	var/flags = 0			
-	
+	var/flags = 0
+
 	/// This var is set to TRUE after the subsystem has been initialized.
 	var/initialized = FALSE
 
 	/// Set to 0 to prevent fire() calls, mostly for admin use or subsystems that may be resumed later
 	///		use the [SS_NO_FIRE] flag instead for systems that never fire to keep it from even being added to list that is checked every tick
 	var/can_fire = TRUE
-	
+
 	///Bitmap of what game states can this subsystem fire at. See [RUNLEVELS_DEFAULT] for more details.
 	var/runlevels = RUNLEVELS_DEFAULT	//points of the game at which the SS can fire
 
 	/*
 	 * The following variables are managed by the MC and should not be modified directly.
 	 */
-	
+
 	/// Last world.time the subsystem completed a run (as in wasn't paused by [MC_TICK_CHECK])
 	var/last_fire = 0
-	
+
 	/// Scheduled world.time for next fire()
 	var/next_fire = 0
-	
+
 	/// Running average of the amount of milliseconds it takes the subsystem to complete a run (including all resumes but not the time spent paused)
-	var/cost = 0			
-	
+	var/cost = 0
+
 	/// Running average of the amount of tick usage in percents of a tick it takes the subsystem to complete a run
-	var/tick_usage = 0		
-	
+	var/tick_usage = 0
+
 	/// Running average of the amount of tick usage (in percents of a game tick) the subsystem has spent past its allocated time without pausing
 	var/tick_overrun = 0
-	
+
 	/// Tracks the current execution state of the subsystem. Used to handle subsystems that sleep in fire so the mc doesn't run them again while they are sleeping
 	var/state = SS_IDLE
-	
+
 	/// Tracks how many fires the subsystem has consecutively paused on in the current run
 	var/paused_ticks = 0
-	
+
 	/// Tracks how much of a tick the subsystem has consumed in the current run
 	var/paused_tick_usage
-	
+
 	/// Tracks how many fires the subsystem takes to complete a run on average.
 	var/ticks = 1
-	
+
 	/// Tracks the amount of completed runs for the subsystem
 	var/times_fired = 0
-	
+
 	/// Time the subsystem entered the queue, (for timing and priority reasons)
-	var/queued_time = 0		
-	
+	var/queued_time = 0
+
 	/// Priority at the time the subsystem entered the queue. Needed to avoid changes in priority (by admins and the like) from breaking things.
-	var/queued_priority 	
-	
+	var/queued_priority
+
 	/// How many times we suspect a subsystem type has crashed the MC, 3 strikes and you're out!
-	var/static/list/failure_strikes 
-	
+	var/static/list/failure_strikes
+
 	/// Next subsystem in the queue of subsystems to run this tick
 	var/datum/controller/subsystem/queue_next
 	/// Previous subsystem in the queue of subsystems to run this tick
 	var/datum/controller/subsystem/queue_prev
 
+	// Does this subsystem process while explosions are going?
+	var/wait_for_explosions = FALSE
```

Comment:
you should just use flags_1 (I think this is at the datum level)

and SS_PAUSEABLE define flag (or maybe SS_INTERRUPTABLE)

Reply:
subsystem's already existing flags would be the more appropriate place to put this.

also user definable subsystem vars should go above these in the definition.